### PR TITLE
No longer need to explicitly include/link KWIVER dirs

### PR DESCRIPTION
**Pull request by mleotta**

### CODE REVIEW

**Review by mwoehlke-kitware — APPROVED** [review]

**Comment by mleotta** [comment]
@mwoehlke-kitware I isolated this change from various others you suggested to confirm that this is the change causing build errors in CI.  For some reason, without these two lines we get weird errors about various classes in PDAL headers having virtual functions but non-virtual destructors.
```
[ 18%] Building CXX object maptk/CMakeFiles/maptk.dir/write_pdal.cxx.o
In file included from /home/travis/build/Kitware/TeleSculptor/maptk/write_pdal.cxx:43:
In file included from /opt/kitware/fletch/include/pdal/PointView.hpp:40:
In file included from /opt/kitware/fletch/include/pdal/PointContainer.hpp:39:
In file included from /opt/kitware/fletch/include/pdal/PointLayout.hpp:44:
In file included from /opt/kitware/fletch/include/pdal/Metadata.hpp:41:
In file included from /opt/kitware/fletch/include/pdal/util/Uuid.hpp:62:
/opt/kitware/fletch/include/pdal/util/Inserter.hpp:42:16: error: 
      'pdal::Inserter' has virtual functions but non-virtual destructor
      [-Werror,-Wnon-virtual-dtor]
class PDAL_DLL Inserter
```
Do you have any idea what would cause this?  It seems the build is finding the right PDAL location.  I'm don't see how changes to the include directories would cause this.  I'm not seeing this issue in a Windows build.

**Comment by mwoehlke-kitware** [comment]
> Do you have any idea what would cause this?

Most likely the PDAL includes have somehow flipped from `-isystem` to `-I`, possibly because `${KWIVER_INCLUDE_DIRS}` had something like `<fletch-install>/include`, which happens to also be `${PDAL_INCLUDE_DIRS}`. Changing the includes in `maptk/CMakeLists.txt` from `PRIVATE` to `PRIVATE SYSTEM` will probably fix it, and should probably be done anyway.

**Comment by mleotta** [comment]
@mwoehlke-kitware I think I made the change as you suggested, but it did not change the error.

**Comment by mwoehlke-kitware** [comment]
Probably somewhere is still including `/opt/kitware/fletch/include` as non-`SYSTEM`. Unfortunately the CI build does not show command lines, so I can't verify that's what's happening, but it seems likely.

**Comment by mleotta** [comment]
One step closer.  It needed to be `SYSTEM PRIVATE` not `PRIVATE SYSTEM` to work properly.  Now I have a new issue on the library linking side to figure out.

**Comment by mwoehlke-kitware** [comment]
Yike... it looks like `PDAL_LIBRARIES` is "pdal" and not the path to the library? (And probably that used to work with `link_directories("${KWIVER_LIBRARY_DIRS}")`.)

**Comment by mleotta** [comment]
@mwoehlke-kitware I have a workaround in place that works across platforms.  The problem seems to be that PDAL (at least version 1.7.2 from Fletch) has issues with its exported targets.  On Windows the primary library name is exported at "pdalcpp" and "pdal_util" is also added to `PDAL_LIBRARIES`.  For other platforms, `PDAL_LIBRARIES` only contains "pdalcpp".  However, on non-Windows platforms it exports a target named "pdal_base" instead of "pdalcpp".  So I'm correcting the `PDAL_LIBRARIES` to use the correct target name if version 1.7.2 and not Windows.

I expect this issue persists for more than 1.7.2, but I don't know if it's been fixed and I haven't tried any other versions yet.

**Comment by mwoehlke-kitware** [comment]
In PDAL 2.2.0, we have:
```cmake
# Note: One should be able to just link libpdalcpp.so/dylib/lib.  But
#   I can't find a way to use pdalcpp because it's not
#   an actual library that cmake knows about.  If you can figure this out,
#   a change would be nice.
set(PDAL_LIBRARIES pdal_base pdal_util)
```

**Review by mwoehlke-kitware — APPROVED** [review]

File: maptk/CMakeLists.txt [hunk 1]
```diff
@@ -47,10 +47,15 @@ option(TELESCULPTOR_USE_PDAL "Enable PDAL support for saving to LAS" ON)
 if(TELESCULPTOR_USE_PDAL)
   find_package(PDAL 1.0.0 REQUIRED)
 
+  # PDAL library names are improperly exported in PDAL 1.7.2 for non-Windows
+  if (${PDAL_VERSION} VERSION_EQUAL "1.7.2" AND NOT WIN32)
+    set(PDAL_LIBRARIES "pdal_base")
+  endif(WIN32)
```
Comment (mleotta) [inline maptk/CMakeLists.txt:53]: Ah, this should just be `endif()`

**Pull request merged**